streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorStateManager.java

```diff
@@ -124,48 +127,94 @@ public void register(StateStore store, StateRestoreCallback stateRestoreCallback
 
         this.stores.put(store.name(), store);
 
+        if (isStandby) {
+            if (store.persistent())
+                restoreCallbacks.put(store.name(), stateRestoreCallback);
+        } else {
+            restore(store, stateRestoreCallback);
+        }
+    }
+
+    private void restore(StateStore store, StateRestoreCallback stateRestoreCallback) {
```

Comment:
nit: maybe rename to "restoreActiveStates" to be consistent with "updateStandbyStates".